Src/Fido2/AuthenticatorAttestationResponse.cs

```diff
@@ -157,6 +169,119 @@ public static AuthenticatorAttestationResponse Parse(AuthenticatorAttestationRaw
 
+        VerifyTrustAnchor(metadataEntry, trustPath);
+
+        // 22. Assess the attestation trustworthiness using the outputs of the verification procedure in step 14, as follows:
+        //     If self attestation was used, check if self attestation is acceptable under Relying Party policy.
+        //     If ECDAA was used, verify that the identifier of the ECDAA-Issuer public key used is included in the set of acceptable trust anchors obtained in step 15.
+        //     Otherwise, use the X.509 certificates returned by the verification procedure to verify that the attestation public key correctly chains up to an acceptable root certificate.
+
+        // Check status resports for authenticator with undesirable status
+        var latestStatusReport = metadataEntry?.GetLatestStatusReport();
+        if (latestStatusReport != null && config.UndesiredAuthenticatorMetadataStatuses.Contains(latestStatusReport.Status))
+        {
+            throw new UndesiredMetdatataStatusFido2VerificationException(latestStatusReport);
+        }
+
+        // 23. Verify that the credentialId is ≤ 1023 bytes.
+        // Handled by AttestedCredentialData constructor
+
+        // 24. Check that the credentialId is not yet registered to any other user. 
+        //     If registration is requested for a credential that is already registered to a different user,
+        //     the Relying Party SHOULD fail this registration ceremony, or it MAY decide to accept the registration, e.g. while deleting the older registration
+
+        if (await isCredentialIdUniqueToUser(new IsCredentialIdUniqueToUserParams(authData.AttestedCredentialData.CredentialID, originalOptions.User), cancellationToken) is false)
+        {
+            throw new Fido2VerificationException(Fido2ErrorCode.NonUniqueCredentialId, Fido2ErrorMessages.NonUniqueCredentialId);
+        }
+
+        // 25. If the attestation statement attStmt verified successfully and is found to be trustworthy,
+        //     then register the new credential with the account that was denoted in the options.user passed to create(), 
+        //     by associating it with the credentialId and credentialPublicKey in the attestedCredentialData in authData,
+        //     as appropriate for the Relying Party's system.
+
+        // 26. If the attestation statement attStmt successfully verified but is not trustworthy per step 16 above,
+        //     the Relying Party SHOULD fail the registration ceremony.
+        //     This implementation throws if the outputs are not trustworthy for a particular attestation type.
+
+        return new AttestationVerificationSuccess
+        {
+            Type = Raw.Type,
+            Id = authData.AttestedCredentialData.CredentialID,
+            PublicKey = authData.AttestedCredentialData.CredentialPublicKey.GetBytes(),
+            SignCount = authData.SignCount,
+            //Transports = result of response.getTransports();
```

Comment:
Why do we not return Transports anymore?

Reply:
We didn't return transports before, in fact we didn't even collect it in the JavaScript at all, I added in [this](https://github.com/passwordless-lib/fido2-net-lib/blob/53caf81da894691ad10eefcce5fd5979b4a8e6ed/Demo/wwwroot/js/custom.register.js#L129) but it looks like it needs to be added to all of the other registration JavaScript files, and collect transports in [AuthenticatorAttestationRawResponse.ResponseData](https://github.com/passwordless-lib/fido2-net-lib/blob/53caf81da894691ad10eefcce5fd5979b4a8e6ed/Src/Fido2.Models/AuthenticatorAttestationRawResponse.cs#L26), and then populate transports in the [controller from the raw response](https://github.com/passwordless-lib/fido2-net-lib/blob/53caf81da894691ad10eefcce5fd5979b4a8e6ed/Demo/Controller.cs#L131).  There are a few reasons to collect transports, to help with user facing dialogs, but also to compare what transports are being used against a policy, or comparing what transports should be available according to the authenticator metadata.  Basically this is incomplete at the moment.